extensions/metadata/interface/src/main/java/org/exist/storage/md/xquery/DocumentByPair.java

```diff
@@ -39,20 +42,36 @@
  */
 public class DocumentByPair extends BasicFunction {
 	
-	private static final QName NAME = new QName("document-by-pair", NAMESPACE_URI, PREFIX);
-	private static final String DESCRIPTION = "Get the documents by match key/value pair.";
-	private static final FunctionReturnSequenceType RETURN = new FunctionReturnSequenceType(Type.ITEM, Cardinality.ZERO_OR_MORE, "Resources which have this key/value pair in their metadata");
+	private static final QName RES = new QName("resource-by-pair", NAMESPACE_URI, PREFIX);
+    private static final QName COL = new QName("collection-by-pair", NAMESPACE_URI, PREFIX);
+    private static final QName DOC = new QName("document-by-pair", NAMESPACE_URI, PREFIX);
+
+    private static final SequenceType[] PARAMS = new SequenceType[] {
+        new FunctionParameterSequenceType("key", Type.STRING, Cardinality.EXACTLY_ONE, "The key to match"),
+        new FunctionParameterSequenceType("value", Type.STRING, Cardinality.EXACTLY_ONE, "The value to match")
+    };
+
+    private static final FunctionReturnSequenceType RETURN = new FunctionReturnSequenceType(Type.ITEM, Cardinality.ZERO_OR_MORE, "Resources which have this key/value pair in their metadata");
 	
     public final static FunctionSignature signatures[] = {
-		new FunctionSignature(
-			NAME,
-			DESCRIPTION,
-			new SequenceType[] { 
-				 new FunctionParameterSequenceType("key", Type.STRING, Cardinality.EXACTLY_ONE, "The key to match"),
-				 new FunctionParameterSequenceType("value", Type.STRING, Cardinality.EXACTLY_ONE, "The value to match")
-			}, 
-			RETURN
-		)
+//		new FunctionSignature(
```

Comment:
Okay if it is unimplemented, please remove it. Better to send a complete feature in a future PR. Otherwise sometimes things hang around for a very long time in an uncompleted fashion